### PR TITLE
Augment describe to search for matching things if it doesn't match the original resource.

### PR DESCRIPTION
**Pull request by brendandburns**
@smarterclayton @jlowdermilk @ghodss @bgrant0607

### CODE REVIEW

File: pkg/kubectl/cmd/describe.go [hunk 1]
```diff
@@ -88,3 +95,39 @@ func RunDescribe(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []s
 	fmt.Fprintf(out, "%s\n", s)
 	return nil
 }
+
+func DescribeMatchingResources(mapper meta.RESTMapper, typer runtime.ObjectTyper, describer kubectl.Describer, f *cmdutil.Factory, namespace, rsrc, prefix string, out io.Writer) error {
+	r := resource.NewBuilder(mapper, typer, f.ClientMapperForCommand()).
+		NamespaceParam(namespace).DefaultNamespace().
+		ResourceTypeOrNameArgs(true, rsrc).
+		SingleResourceType().
+		Do()
```
Comment (smarterclayton) [inline pkg/kubectl/cmd/describe.go]: Would be nice if you could reuse the builder (to guard against bit rot) but it's not easy to do here I don't think.

File: pkg/kubectl/cmd/describe.go [hunk 2]
```diff
@@ -88,3 +95,39 @@ func RunDescribe(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []s
...
+	obj, err := r.Object()
+	if err != nil {
+		return err
+	}
+	objs, err := f.List(obj)
```
Comment (smarterclayton) [inline pkg/kubectl/cmd/describe.go]: You shouldn't need this, the object can be genetically treated as a list by the resource builder.  Add Flatten to the builder and invoke Infos() or the Visit on the result.  Infos has name already (or should).

Comment (smarterclayton) [inline pkg/kubectl/cmd/describe.go]: Infos also has namespace set

**Comment by bgrant0607** [comment]
Fixes #5905. Glad to have it, but not in v1.0, I'll point out.

File: pkg/kubectl/cmd/describe.go (hunk 1 above)
Comment (brendandburns) [inline pkg/kubectl/cmd/describe.go]: Acknowledged.

File: pkg/kubectl/cmd/describe.go (hunk 2 above)
Comment (brendandburns) [inline pkg/kubectl/cmd/describe.go]: Cool, done.

**Comment by brendandburns** [comment]
@smarterclayton comments addressed, please take another look.

@bgrant0607 mea culpa, I think this is important for the end user experience when using ```kubectl run-container ...``` I should have filed an isssue for that (or elevated the priority of #5905)

Closes #5905

**Comment by smarterclayton** [comment]
LGTM once travis is green

**Comment by smarterclayton** [comment]
Actually, can you add a test for this code path to hack/test-cmd?

**Pull request merged**